test/dynamo/test_repros.py

```diff
@@ -1959,6 +1959,18 @@ def f(x, y):
             self.assertEqual(f(x, x), opt_f(x, x))
             self.assertEqual(f(x, y), opt_f(x, y))
 
+    def test_output_aliases_intermediate(self):
+        def f(x):
+            intermediate = x.mul(2)
+            return intermediate.view(-1)
+
+        opt_f = torch._dynamo.optimize("aot_eager")(f)
+
+        for b in [True, False]:
+            x = torch.randn(4, requires_grad=b)
+            self.assertEqual(f(x), opt_f(x))
+            self.assertEqual(f(x), opt_f(x))
```

Comment:
Did you want to assert that the output is view or something?

Reply:
yep, adding